website/versioned_docs/version-5.13.0/standardized-functions.md

```diff
@@ -1,5 +1,223 @@
+
+1. **Asset Class**: The category of the derivative (e.g., credit, equity, commodity) that influences the margin rate.
+2. **Product Class**: A specific classification of the financial product, aligned with ISDA guidelines to ensure standardization.
+3. **Notional Amount**: The total value of the derivative contract, representing the size of the exposure.
+4. **Notional Currency**: The currency in which the notional amount is denominated, necessary for financial calculations.
+5. **Duration (in years)**: The remaining time until the derivative contract matures, which impacts the risk profile and margin requirement.
+
+Using the **concept of product class** helps ensure consistency in determining the notional amount and duration for each type of product. This standardization is vital for harmonizing calculations across different market participants and enhancing comparability.
+
+## Extraction of Information
+
+Function `BuildStandardizedSchedule ` is designed to **extract all relevant trade information** necessary for calculating the gross initial margin. It streamlines the process of gathering data from trades, ensuring accuracy and efficiency.
+
+``` Haskell
+func BuildStandardizedSchedule:
+    inputs: 
+        trade Trade (1..1)
+    output: standardizedSchedule StandardizedSchedule (1..1)
+    alias assetClass:
+        StandardizedScheduleAssetClass(trade)
+    alias productClass:
+        StandardizedScheduleProductClass(trade)
+    set standardizedSchedule -> assetClass:
+        assetClass
+    set standardizedSchedule -> productClass:
+        productClass
+    set standardizedSchedule -> notional:
+        StandardizedScheduleNotional(trade, assetClass, productClass)
+    set standardizedSchedule -> notionalCurrency:
+        StandardizedScheduleNotionalCurrency(trade, assetClass, productClass)
+    set standardizedSchedule -> durationInYears:
+        StandardizedScheduleDuration(trade, assetClass, productClass)
+```
+
+The function operates by extracting critical data points from the provided trade information. This includes identifying the asset class and product class, which determine the applicable margin rates. Additionally, the function collects the notional amount and currency, as well as the duration of the contract.
+
+This information is then organized into a standardized schedule, ensuring consistency and clarity for subsequent calculations.
+
+## Calculation of Initial Margin Amounts
+
+The next step involves **determining the required margin rate** for the trade based on its asset class and duration using `GetStandardizedScheduleMarginRate` function:
+
+``` Haskell
+func GetStandardizedScheduleMarginRate:
+    inputs: 
+        assetClass StandardizedScheduleAssetClassEnum (1..1)
+        durationInYears number (1..1)
+    output: percentage number (1..1)
+    set percentage:
```

Comment:
Same as previous comment, it's nice to see the logic but really users should go into a modelling tool to review it

I won't add this comment to the other functions but its the same thing